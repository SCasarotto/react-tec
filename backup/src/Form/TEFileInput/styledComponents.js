import styled from 'styled-components'

import { FaFileUpload, FaRegWindowClose } from 'react-icons/fa'

import TEButton from './../TEButton'

export const Wrapper = styled.div`
	display: flex;
	flex-direction: column;
	width: 100%;
`
export const InputWrapper = styled.div`
	display: flex;
	justify-content: space-between;
	align-items: center;
	width: 100%;
	background-color: ${(props) => props.theme.white};
	border-radius: 5px;
	padding-right: 5px;

	${(props) => {
		const { theme, disabled, active } = props
		if (active) {
			return `
                border: 1px solid ${theme.primary};
                box-shadow: 0 0 0 1px ${theme.primary} inset;
            `
		} else if (disabled) {
			return `
                border: 1px solid ${theme.lighterGray};
                box-shadow: none;
            `
		}
		return `
			border: 1px solid ${theme.lightGray};
			box-shadow: none;
		`
	}}

	transition: border-color 0.2s ease-in, box-shadow 0.2s ease-in;
	:hover,
	:active {
		border: 1px solid ${(props) => props.theme.primary};
		box-shadow: 0 0 0 1px ${(props) => props.theme.primary} inset;
	}
`
export const Label = styled.label`
	display: block;
	width: 100%;
	font-size: 14px;
	color: ${(props) => (props.disabled ? props.theme.gray : props.theme.darkGray)};
	padding: 10px;

	${(props) => {
		const { disabled, active } = props
		if (disabled) {
			return 'cursor: not-allowed'
		} else if (active) {
			return 'cursor: copy'
		}
		return 'cursor: pointer'
	}}
`

export const LabelCopy = styled.span``

export const LabelIcon = styled(FaFileUpload)`
	margin-right: 10px;
	vertical-align: middle;
	font-size: 18px;
`

export const Input = styled.input`
	display: none;
`

export const ClearButton = styled(TEButton)`
	width: auto;
	padding: 5px 10px;
	text-align: center;
`

export const ButtonIcon = styled(FaRegWindowClose)`
	color: inherit;
	display: block;
`

export const ErrorMessage = styled.span`
	display: block;
	padding-top: 5px;
	padding-bottom: 5px;
	font-size: 14px;
	color: ${(props) => props.theme.errorRed};
`
