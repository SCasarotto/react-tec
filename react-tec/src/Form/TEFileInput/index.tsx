import React, { useState, ChangeEvent } from 'react'

import {
	Wrapper,
	InputWrapper,
	Label,
	LabelCopy,
	LabelIcon,
	Input,
	ClearButton,
	ButtonIcon,
	ErrorMessage,
} from './styledComponents'

export interface TEFileInputProps
	extends Omit<
		React.DetailedHTMLProps<
			React.InputHTMLAttributes<HTMLInputElement>,
			HTMLInputElement
		>,
		'onChange'
	> {
	//TODO: Fix this. styled-components is weird with refs
	ref?: any
	onChange?(files: FileList): void
	placeholder?: string
	hideClearButton?: boolean
	filePattern?: RegExp
	resetKey?: string
}
export const TEFileInput: React.FC<TEFileInputProps> = (props) => {
	const [active, setActive] = useState(false)
	const [fileArray, setFileArray] = useState<FileList>(new FileList())
	const [errorData, setErrorData] = useState({ error: false, message: '' })
	const [inputKey, setInputKey] = useState<string | number>(
		props.resetKey || 'inputKey',
	)

	const onFileChange = (files: FileList | null) => {
		const { filePattern, onChange } = props

		if (!files) {
			console.error('No file provided to input.')
			return
		}

		//Validate Files
		if (filePattern && filePattern instanceof RegExp) {
			for (let i = 0; i < files.length; i++) {
				if (!files[i].type.match(filePattern)) {
					console.warn(
						'One of the selected files is not of the correct type.',
					)
					setErrorData({
						error: true,
						message:
							'One of the selected files is not of the correct type.',
					})
					return
				}
			}
		}

		setFileArray(files)
		setErrorData({
			error: false,
			message: '',
		})
		if (onChange) {
			onChange(files)
		}
	}
	const onDragEnter = () => setActive(true)
	const onDragLeave = () => setActive(false)
	const onDragOver = (e: React.DragEvent<HTMLDivElement>) =>
		e.preventDefault()
	const onDrop = (e: React.DragEvent<HTMLDivElement>) => {
		e.preventDefault()
		setActive(false)
		onFileChange(e.dataTransfer.files)
	}
	const handleClearInput = () => {
		setInputKey(new Date().getTime()) //This will reset the input
		setFileArray(new FileList())
		setErrorData({
			error: false,
			message: '',
		})
	}

	const {
		disabled,
		className = '',
		placeholder = 'Choose a file...',
		hideClearButton = false,
		...rest
	} = props

	let labelCopy = placeholder
	if (fileArray && fileArray.length === 1) {
		labelCopy = fileArray[0].name
	} else if (fileArray && fileArray.length > 1) {
		labelCopy = `${fileArray.length} files selected`
	}

	return (
		<Wrapper className={`TEFileInput ${className}`}>
			<InputWrapper
				onDragEnter={onDragEnter}
				onDragLeave={onDragLeave}
				onDragOver={onDragOver}
				onDrop={onDrop}
				active={active}
				disabled={disabled}
				className="TEFileInputInputWrapper">
				<Label
					active={active}
					disabled={disabled}
					className="TEFileInputLabel">
					<LabelCopy className="TEFileInputCopy">
						<LabelIcon className="TEFileInputIcon" />
						{labelCopy}
					</LabelCopy>
					<Input
						{...rest}
						disabled={disabled}
						type="file"
						key={inputKey}
						onChange={(e: ChangeEvent<HTMLInputElement>) =>
							onFileChange(e.target.files)
						}
						className="TEFileInputInput"
					/>
				</Label>
				{!hideClearButton && (
					<ClearButton
						onClick={handleClearInput}
						disabled={disabled}
						className="TEFileInputClearButton">
						<ButtonIcon className="TEFileInputClearIcon" />
					</ClearButton>
				)}
			</InputWrapper>
			{errorData && errorData.error && (
				<ErrorMessage className="TEFileInputErrorMessage">
					{errorData.message}
				</ErrorMessage>
			)}
		</Wrapper>
	)
}
