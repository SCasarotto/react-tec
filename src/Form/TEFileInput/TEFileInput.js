//
// Version 0.3.0
//

import React, { useState } from 'react'
import PropTypes from 'prop-types'

import {
	Wrapper,
	InputWrapper,
	Label,
	LabelCopy,
	LabelIcon,
	Input,
	ClearButton,
	ButtonIcon,
	ErrorMessage,
} from './styledComponents'

const TEFileInput = (props) => {
	const [active, setActive] = useState(false)
	const [fileArray, setFileArray] = useState([])
	const [errorData, setErrorData] = useState({ error: false, message: '' })
	const [inputKey, setInputKey] = useState(props.resetKey || 'inputKey')

	const onFileChange = (e, droppedFiles) => {
		const { pattern, onChange } = props
		const files = e.target.files || droppedFiles

		if (!files) {
			console.error('No file provided to input.')
			return
		}

		//Validate Files
		if (pattern && pattern instanceof RegExp) {
			for (let i = 0; i < files.length; i++) {
				if (!files[i].type.match(pattern)) {
					console.warn('One of the selected files is not of the correct type.')
					setErrorData({
						error: true,
						message: 'One of the selected files is not of the correct type.',
					})
					return
				}
			}
		}

		setFileArray(files)
		setErrorData({
			error: false,
			message: '',
		})
		onChange(files)
	}
	const onDragEnter = (e) => setActive(true)
	const onDragLeave = (e) => setActive(false)
	const onDragOver = (e) => e.preventDefault()
	const onDrop = (e) => {
		e.preventDefault()
		setActive(false)
		onFileChange(e, e.dataTransfer.files)
	}
	const handleClearInput = () => {
		setInputKey(new Date().getTime()) //This will reset the input
		setFileArray([])
		setErrorData({
			error: false,
			message: '',
		})
	}

	const { disabled, className, placeholder = 'Choose a file...', ...rest } = props

	let labelCopy = placeholder
	if (fileArray && fileArray.length === 1) {
		labelCopy = fileArray[0].name
	} else if (fileArray && fileArray.length > 1) {
		labelCopy = `${fileArray.length} files selected`
	}

	return (
		<Wrapper className={className}>
			<InputWrapper
				onDragEnter={onDragEnter}
				onDragLeave={onDragLeave}
				onDragOver={onDragOver}
				onDrop={onDrop}
				active={active}
				disabled={disabled}
				className='TEFileInputInputWrapper'
			>
				<Label active={active} disabled={disabled} className='TEFileInputLabel'>
					<LabelCopy className='TEFileInputCopy'>
						<LabelIcon className='TEFileInputIcon' />
						{labelCopy}
					</LabelCopy>
					<Input
						{...rest}
						disabled={disabled}
						type='file'
						key={inputKey}
						onChange={onFileChange}
						className='TEFileInputInput'
					/>
				</Label>
				<ClearButton
					onClick={handleClearInput}
					disabled={disabled}
					className='TEFileInputClearButton'
				>
					<ButtonIcon className='TEFileInputClearIcon' />
				</ClearButton>
			</InputWrapper>
			{errorData && errorData.error && (
				<ErrorMessage className='TEFileInputErrorMessage'>{errorData.message}</ErrorMessage>
			)}
		</Wrapper>
	)
}

TEFileInput.propTypes = {
	onChange: PropTypes.func.isRequired,
	accept: PropTypes.string, //OS Hint of file type for dialog - NOT safety check of type
	// pattern: PropTypes.RegExp, //This is the actual strict catch. Regex // Don't know regex proptype
	disabled: PropTypes.bool,
}

export default TEFileInput
