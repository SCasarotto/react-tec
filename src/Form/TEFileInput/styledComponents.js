import styled from 'styled-components'

import { FaFileUpload, FaRegWindowClose } from 'react-icons/fa'

import TEButton from './../TEButton'

export const Wrapper = styled.div`
	${(props) => {
		return `
            display: flex;
            flex-direction: column;
            width: 100%;
        `
	}}
`
export const InputWrapper = styled.div`
	${(props) => {
		const { theme, disabled, active } = props
		let styles = `
            display: flex;
            justify-content: space-between;
            align-items: center;
            width: 100%;
            background-color: ${theme.white};
            border-radius: 5px;
            padding-right: 5px;

            transition: border-color 0.2s ease-in, box-shadow 0.2s ease-in;
        `

		if (disabled) {
			styles += `
                border: 1px solid ${theme.lighterGray};
                box-shadow: none;
            `
		} else if (active) {
			styles += `
                border: 1px solid ${theme.primary};
                box-shadow: 0 0 0 1px ${theme.primary} inset;
            `
		} else {
			styles += `
                border: 1px solid ${theme.lightGray};
                box-shadow: none;

                :hover {
                    border: 1px solid ${theme.primary};
                    box-shadow: 0 0 0 1px ${theme.primary} inset;
                }
                :active {
                    border: 1px solid ${theme.primary};
                    box-shadow: 0 0 0 1px ${theme.primary} inset;
                }
            `
		}

		return styles
	}}
`
export const Label = styled.label`
	${(props) => {
		const { theme, disabled, active } = props
		let cursor = 'pointer'
		if (disabled) {
			cursor = 'not-allowed'
		} else if (active) {
			cursor = 'copy'
		}

		return `
            display: block;
            width: 100%;
            font-size: 14px;
            color: ${disabled ? theme.gray : theme.darkGray};
            padding: 10px;
            cursor: ${cursor};
        `
	}}
`

export const LabelCopy = styled.span`
	${(props) => {
		return `

        `
	}}
`

export const LabelIcon = styled(FaFileUpload)`
	${(props) => {
		return `
            margin-right: 10px;
            vertical-align: middle;
            font-size: 18px;
        `
	}}
`

export const Input = styled.input`
	${(props) => {
		return `
            display: none;
        `
	}}
`

export const ClearButton = styled(TEButton)`
	${(props) => {
		return `
            width: auto;
            padding: 5px 10px;
            text-align: center;
        `
	}}
`

export const ButtonIcon = styled(FaRegWindowClose)`
	${(props) => {
		return `
            color: inherit;
            display: block;
        `
	}}
`

export const ErrorMessage = styled.span`
	${(props) => {
		const { theme } = props
		return `
            display: block;
            padding-top: 5px;
            padding-bottom: 5px;
            font-size: 14px;
            color: ${theme.errorRed};
        `
	}}
`
