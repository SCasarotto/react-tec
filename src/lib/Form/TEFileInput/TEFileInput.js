//
// Version 0.3.0
//

import React, { useState } from 'react'
import PropTypes from 'prop-types'

import {
    Wrapper,
    InputWrapper,
    Label,
    LabelCopy,
    LabelIcon,
    Input,
    ClearButton,
    ButtonIcon,
    ErrorMessage,
} from './styledComponents'

const TEFileInput = (props) => {
    const [active, setActive] = useState(false)
    const [fileArray, setFileArray] = useState([])
    const [errorData, setErrorData] = useState({ error: false, message: '' })
    const [inputKey, setInputKey] = useState(props.resetKey || 'inputKey')

    const onFileChange = (e, droppedFiles) => {
        const { pattern, onChange } = props
        const files = e.target.files || droppedFiles

        if (!files) {
            console.error('No file provided to input.')
            return
        }

        //Validate Files
        if (pattern && pattern instanceof RegExp) {
            for (let i = 0; i < files.length; i++) {
                if (!files[i].type.match(pattern)) {
                    console.warn('One of the selected files is not of the correct type.')
                    setErrorData({
                        error: true,
                        message: 'One of the selected files is not of the correct type.',
                    })
                    return
                }
            }
        }

        setFileArray(files)
        setErrorData({
            error: false,
            message: '',
        })
        onChange(files)

        // const readFile = (file) => {
        //     const { name } = file
        //     const reader = new FileReader()

        //     reader.onload = (e) => {
        //         const { fileArray } = this.state
        //         const updateFileArray = [...fileArray, { name, value: reader.result }]
        //         this.setState({
        //             fileArray: updateFileArray,
        //             error: false,
        //             errorMessage: '',
        //         })
        //         onChange(updateFileArray)
        //     }

        //     reader.readAsDataURL(file)
        // }

        // for (let i = 0; i < files.length; i++) {
        //     readFile(files[i])
        // }
    }
    const onDragEnter = (e) => setActive(true)
    const onDragLeave = (e) => setActive(false)
    const onDragOver = (e) => e.preventDefault()
    const onDrop = (e) => {
        e.preventDefault()
        setActive(false)
        onFileChange(e, e.dataTransfer.files)
    }
    const handleClearInput = () => {
        setInputKey(new Date().getTime()) //This will reset the input
        setFileArray([])
        setErrorData({
            error: false,
            message: '',
        })
    }

    const { inputRef, disabled, className, ...rest } = props

    let labelCopy = 'Choose a file...'
    if (fileArray && fileArray.length === 1) {
        labelCopy = fileArray[0].name
    } else if (fileArray && fileArray.length > 1) {
        labelCopy = `${fileArray.length} files selected`
    }

    return (
        <Wrapper className={className}>
            <InputWrapper
                onDragEnter={onDragEnter}
                onDragLeave={onDragLeave}
                onDragOver={onDragOver}
                onDrop={onDrop}
                active={active}
                disabled={disabled}
            >
                <Label active={active} disabled={disabled}>
                    <LabelCopy>
                        <LabelIcon />
                        {labelCopy}
                    </LabelCopy>
                    <Input
                        {...rest}
                        disabled={disabled}
                        type="file"
                        key={inputKey}
                        onChange={onFileChange}
                    />
                </Label>
                <ClearButton onClick={handleClearInput} disabled={disabled}>
                    <ButtonIcon />
                </ClearButton>
            </InputWrapper>
            {errorData && errorData.error && <ErrorMessage>{errorData.message}</ErrorMessage>}
        </Wrapper>
    )
}

TEFileInput.propTypes = {
    style: PropTypes.object,
    onChange: PropTypes.func.isRequired,
    accept: PropTypes.string, //OS Hint of file type for dialog - NOT safety check of type
    // pattern: PropTypes.RegExp, //This is the actual strict catch. Regex // Don't know regex proptype
    id: PropTypes.string,
    disabled: PropTypes.bool,
}

export default TEFileInput
