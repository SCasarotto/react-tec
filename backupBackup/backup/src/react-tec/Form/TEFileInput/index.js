import TEFileInput from './TEFileInput';

export default TEFileInput;
