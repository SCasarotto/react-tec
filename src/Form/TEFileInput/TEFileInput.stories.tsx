import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TEFileInput, TEFileInputProps } from './TEFileInput'

export default {
	title: 'Form/TEFileInput',
	component: TEFileInput,
	argTypes: {},
} as Meta

const Template: Story<TEFileInputProps> = (args) => <TEFileInput {...args} />

export const Default = Template.bind({})
Default.args = {}
