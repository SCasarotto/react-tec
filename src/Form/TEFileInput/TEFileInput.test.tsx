import React from 'react'
import ReactDOM from 'react-dom'
import { TEFileInput } from './TEFileInput'

describe('TEFileInput', () => {
	it('renders without crashing', () => {
		const div = document.createElement('div')
		ReactDOM.render(<TEFileInput />, div)
		ReactDOM.unmountComponentAtNode(div)
	})
})
