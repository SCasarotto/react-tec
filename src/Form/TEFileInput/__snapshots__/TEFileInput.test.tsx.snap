// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEFileInput matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gtssRu dbxvrr TEFileInput "
  >
    <div
      class="sc-dlnjPT eaxhob TEFileInputInputWrapper"
    >
      <label
        class="sc-hKFyIo boUedS TEFileInputLabel"
      >
        <span
          class="sc-eCApGN TEFileInputCopy"
        >
          <svg
            class="sc-jSFkmK ljtlkI TEFileInputIcon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 384 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm65.18 216.01H224v80c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16v-80H94.82c-14.28 0-21.41-17.29-11.27-27.36l96.42-95.7c6.65-6.61 17.39-6.61 24.04 0l96.42 95.7c10.15 10.07 3.03 27.36-11.25 27.36zM377 105L279.1 7c-4.5-4.5-10.6-7-17-7H256v128h128v-6.1c0-6.3-2.5-12.4-7-16.9z"
            />
          </svg>
          Choose a file...
        </span>
        <input
          class="sc-gKAblj jUZZoM TEFileInputInput"
          type="file"
        />
      </label>
      <button
        class="sc-bdnylx ccWmoN TEButton sc-iCoHVE iUWVMl TEFileInputClearButton"
        type="button"
      >
        <svg
          class="sc-fujyUd kleyEx TEFileInputClearIcon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm0 394c0 3.3-2.7 6-6 6H54c-3.3 0-6-2.7-6-6V86c0-3.3 2.7-6 6-6h404c3.3 0 6 2.7 6 6v340zM356.5 194.6L295.1 256l61.4 61.4c4.6 4.6 4.6 12.1 0 16.8l-22.3 22.3c-4.6 4.6-12.1 4.6-16.8 0L256 295.1l-61.4 61.4c-4.6 4.6-12.1 4.6-16.8 0l-22.3-22.3c-4.6-4.6-4.6-12.1 0-16.8l61.4-61.4-61.4-61.4c-4.6-4.6-4.6-12.1 0-16.8l22.3-22.3c4.6-4.6 12.1-4.6 16.8 0l61.4 61.4 61.4-61.4c4.6-4.6 12.1-4.6 16.8 0l22.3 22.3c4.7 4.6 4.7 12.1 0 16.8z"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
